src/Microsoft.Azure.SignalR.Management/ServiceHubContextBuilder.cs

```diff
@@ -9,121 +9,53 @@
-            return this;
-        }
-
-        /// <summary>
-        /// Uses Newtonsoft.Json library to serialize messages sent to SignalR.
-        /// </summary>
-        /// <returns>The <see cref="ServiceHubContextBuilder"/> instance itself.</returns>
-        public ServiceHubContextBuilder WithNewtonsoftJsonHubProtocol()
-        {
-            return WithNewtonsoftJsonHubProtocol(o => { });
-        }
-
-        /// <summary>
-        /// Provides a hook to configure services before building.
-        /// </summary>
-        internal ServiceHubContextBuilder ConfigureServices(Action<IServiceCollection> configureAction)
-        {
-            _configureAction = configureAction;
+            _hostBuilder.ConfigureServices(configure);
             return this;
         }
 
         /// <summary>
         /// Builds <see cref="ServiceHubContext"/> instances.
         /// </summary>
         /// <returns>The instance of the <see cref="IServiceManager"/>.</returns>
-        public async Task<ServiceHubContext> CreateAsync(string hubName, CancellationToken cancellationToken)
+        internal async Task<ServiceHubContext> CreateAsync(string hubName, CancellationToken cancellationToken)
         {
-            //add requried services
-            using var serviceProvider = _services.BuildServiceProvider();
-            var transportType = serviceProvider.GetRequiredService<IOptions<ServiceManagerOptions>>().Value.ServiceTransportType;
-            var services = new ServiceCollection().Add(_services);
-            services.AddHub(hubName, transportType);
-            _configureAction?.Invoke(services);
-            services.AddSingleton(services.ToList() as IReadOnlyCollection<ServiceDescriptor>);
-
-            //build
-            var serviceHubContext = services.BuildServiceProvider()
-                .GetRequiredService<ServiceHubContextImpl>();
+            _hostBuilder.ConfigureServices(services =>
+            {
+                services.AddHub(hubName);
+                services.AddSingleton(services.ToList() as IReadOnlyCollection<ServiceDescriptor>);
+            });
 
-            //initialize
-            var connectionContainer = serviceHubContext.ServiceProvider.GetService<IServiceConnectionContainer>();
-            if (connectionContainer != null)
+            IHost host = null;
+            try
             {
-                await connectionContainer.ConnectionInitializedTask.OrTimeout(cancellationToken);
+                host = _hostBuilder.Build();
+                await host.StartAsync(cancellationToken);
+                return host.Services.GetRequiredService<ServiceHubContextImpl>();
             }
-
-            if (cancellationToken.IsCancellationRequested)
+            catch
             {
-                await serviceHubContext?.DisposeAsync();
-                return null;
+                if (host != null)
+                {
+                    await host.StopAsync();
+                    host.Dispose();
```

Comment:
`using` host?

Reply:
If no error is thrown, then the host should keep alive. So we cannot `using` host, otherwise the host will always be disposed.

Reply:
using host in catch since StopAsync can throw?

Reply:
We can only use `using` when declare a variable? I added a finally block.